system/Database/BaseConnection.php

```diff
@@ -837,7 +837,7 @@ abstract protected function _transRollback(): bool;
     /**
      * Returns a non-shared new instance of the query builder for this connection.
      *
-     * @param array|string $tableName
+     * @param array|BaseBuilder|Closure|string $tableName
```

Comment:
This PR seems to be good, but the method name `table()` and the variable name `$tableName` don't make sense to me.

Reply:
What do you mean?

Reply:
@kenjis changing parameter name is bc break per php 8.0, ref https://3v4l.org/hpqdm#v8.0.14 vs https://3v4l.org/Gtdk2#v8.0.14

The current change is ok for me

Reply:
Originally, I was uncomfortable with the name of this table() method. However, the table() method barely made sense since it specified a table name.
I wondered if we could change the name  (add a new method) to a better name if it could specify a subquery.

If no one is uncomfortable with `table()`, I won't insist on it.

Reply:
Can you suggest your own version of the method name?
I will say right away that the names of ``subquery`` and ``fromSubquery`` are not correct. In the first case, we get not a subquery, but the main query with the inclusion of a subquery. In the second case, it is associated with the QueryBuilder.

Reply:
`createBuilder()` or `newBuilder()`.

Reply:
I would agree with such method names, but I associate them with creating a new/clean instance, and now this is impossible due to the fact that the class constructor requires a table name.
For example, in Laravel, the DB::table() method takes a string, a closure, a builder in the $ table parameter. That is, in this case it does not differ from my version. Except for the array. One has to scatter a rake.
But there is also an undocumented DB::query() method that returns a clean instance of the QB.
Unfortunately, our implementation already uses the query() method for a raw request. And the inability to create a clean instance of QB is a headache.

Reply:
Laravel has such `table()` method, then it might be not so bad to use `table()`. Because CI4 has many similar methods with Laravel.
And how about that `$db->table(null)` returns a new/clean QB instance?
If QB contructor's `$tableName` accepts null, we can do it.

Reply:
Alternatively, it's interesting.
But in fact, the ``table()`` method is an alias for the ``from()`` method.
And then the construction ``$db->table(null)->from()`` looks a little strange )